hw3/season06/_matrix.cpp

```diff
@@ -0,0 +1,253 @@
+#include <iostream>
+#include <vector>
+#include <stdexcept>
+#include <mkl.h>
+
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+#include <pybind11/numpy.h>
+#include <pybind11/operators.h>
+
+using namespace std;
+namespace py = pybind11;
+
+class Matrix
+{
+public:
+    // default constructor
+    Matrix() : _rows(0), _cols(0) {};
```

Comment:
This is fine, and you probably already know it: https://stackoverflow.com/questions/228783/what-are-the-rules-about-using-an-underscore-in-a-c-identifier .

This use does not violate the standard and you do not need to change for a homework implementation.  But I would like to remind that it's not a bad idea to avoid a leading underscore in production code when there is another choice.